src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/AlertSetting/SetAzureRmRecoveryServicesAsrAlertSetting.cs

```diff
@@ -0,0 +1,172 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Threading;
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Sets Azure Site Revcovery Notification / Alert.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Set,
+        "AzureRmRecoveryServicesAsrAlertSetting",
+        DefaultParameterSetName = ASRParameterSets.Set)]
+    [Alias(
+        "Set-ASRNotificationSetting",
+        "Set-AzureRmRecoveryServicesAsrNotificationSetting",
+        "Set-ASRAlertSetting")]
+    [OutputType(typeof(ASRAlertSetting))]
+    public class SetAzureRmRecoveryServicesAsrAlertSetting : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets EmailSubscriptionOwner .Mail to subscription owner.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.EmailToSubscriptionOwner, Mandatory = true)]
+        public SwitchParameter EmailSubscriptionOwners { get; set; }
+
+        [Parameter(ParameterSetName = ASRParameterSets.DisableEmailToSubcriptionOwner, Mandatory = true)]
+        public SwitchParameter DisableEmailToSubscriptionOwners { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the custom email list.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.Set, Mandatory = false)]
+        [Parameter(ParameterSetName = ASRParameterSets.DisableEmailToSubcriptionOwner)]
+        [Parameter(ParameterSetName = ASRParameterSets.EmailToSubscriptionOwner)]
+        [ValidateNotNullOrEmpty]
+        public string[] CustomEmailAddress { get; set; }
+
+        /// <summary>
+        ///     Gets or sets locale.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.Set, Mandatory = false)]
+        [Parameter(ParameterSetName = ASRParameterSets.DisableEmailToSubcriptionOwner)]
+        [Parameter(ParameterSetName = ASRParameterSets.EmailToSubscriptionOwner)]
+        [ValidateNotNullOrEmpty]
+        public string LocaleID { get; set; }
+
+        /// <summary>
+        ///     Gets or sets switch parameter to disable notification.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.Disable, Mandatory = true)]
+        public SwitchParameter DisableNotification { get; set; }
+
+        /// <summary>
+        ///     ProcessRecord of the command.
+        /// </summary>
+        public override void ExecuteSiteRecoveryCmdlet()
+        {
+            base.ExecuteSiteRecoveryCmdlet();
+            if (this.ShouldProcess(
+                "Alerts",
+                VerbsCommon.Set))
+            {
+                base.ExecuteSiteRecoveryCmdlet();
+                if (this.ShouldProcess(
```

Comment:
@viverm why is there two calls to `this.ShouldProcess` and `base.ExecuteSiteRecoveryCmdlet`?

Reply:
removed